app/lib/package/api_export/exported_api.dart

```diff
@@ -98,40 +98,37 @@ final class ExportedApi {
   Future<void> _gcPrefix(String prefix, Set<String> allPackageNames) async {
     _log.info('Garbage collecting "$prefix"');
 
-    await _listBucket(prefix: prefix + '/api/packages/', delimiter: '/',
-        (item) async {
-      final String packageName;
-      if (item.isObject) {
-        assert(!item.name.endsWith('/'));
-        packageName = item.name.split('/').last;
-      } else {
-        assert(item.name.endsWith('/'));
-        packageName = item.name.without(suffix: '/').split('/').last;
-      }
-      if (!allPackageNames.contains(packageName)) {
-        final info = await _bucket.info(item.name);
-        if (info.updated.isBefore(clock.agoBy(_minGarbageAge))) {
-          // Only delete the item if it's older than _minGarbageAge
-          // This avoids any races where we delete files we've just created
-          await package(packageName).delete();
-        }
+    final gcFilesBefore = clock.agoBy(_minGarbageAge);
+
+    final pkgPrefix = '$prefix/api/packages/';
+    await _listBucket(prefix: pkgPrefix, delimiter: '', (entry) async {
+      final item = switch (entry) {
+        final BucketDirectoryEntry _ => throw AssertionError('unreachable'),
+        final BucketObjectEntry item => item,
+      };
+      final packageName = item.name.without(prefix: pkgPrefix).split('/').first;
+      if (!allPackageNames.contains(packageName) &&
+          item.updated.isBefore(gcFilesBefore)) {
+        // Only delete the item if it's older than _minGarbageAge
+        // This avoids any races where we delete files we've just created
+        // TODO: Conditionally deletion API from package:gcloud would be better!
+        await _bucket.tryDelete(item.name);
```

Comment:
Nit: should we log the fact of deletion (here and elsewhere)? Maybe for a cursory later verification that we are not doing something stupid...